code/modules/antagonists/blob/overmind.dm

```diff
@@ -293,12 +312,12 @@ GLOBAL_LIST_EMPTY(blob_nodes)
 	var/message_a = say_quote(message)
 	var/rendered = span_big("<font color=\"#EE4000\"><b>\[Blob Telepathy\] [name](<font color=\"[blobstrain.color]\">[blobstrain.name]</font>)</b> [message_a]</font>")
 
-	for(var/mob/M in GLOB.mob_list)
-		if(isovermind(M) || isblobmonster(M))
-			to_chat(M, rendered)
-		if(isobserver(M))
-			var/link = FOLLOW_LINK(M, src)
-			to_chat(M, "[link] [rendered]")
+	for(var/mob/creature in GLOB.mob_list)
+		if(HAS_TRAIT(creature, TRAIT_BLOB_ALLY))
+			to_chat(creature, rendered)
+		if(isobserver(creature))
+			var/link = FOLLOW_LINK(creature, src)
+			to_chat(creature, "[link] [rendered]")
```

Comment:
golley this is rather expensive to iterate every goddamn mob in the world for a trait-check... uhhhhhh could we change this to be a global signal with `rendered` as a string payload instead, and have the SIGNAL_HANDLER proc on stuff with the component and whatever else? this would vastly cut down on the number of needless list-accesses on mobs (especially if someone decides to spawn the crab rocket or something). this is sorta the case signals are made for tbh

we could also cut out the isobserver stuff with something like `deadchat_broadcast()` or whatever analogue is suitable. if you don't want to do this then just add a `CHECK_TICK` but i think now is a good time to fix this especially since the new `HAS_TRAIT()` check is definitely more expensive than the typechecks we were doing earlier